Plugin is not working anymore

Plugin fails to load. Here is the log that it spits out:

```
2026-04-03 22:52:01 CDT [pool-3-thread-1] WARN  n.r.c.e.ExternalPluginManager - Unable to start or load external plugin "stealing-artefacts"
java.lang.reflect.InvocationTargetException: null
	at java.desktop/java.awt.EventQueue.invokeAndWait(Unknown Source)
	at java.desktop/java.awt.EventQueue.invokeAndWait(Unknown Source)
	at java.desktop/javax.swing.SwingUtilities.invokeAndWait(Unknown Source)
	at net.runelite.client.externalplugins.ExternalPluginManager.refreshPlugins(ExternalPluginManager.java:333)
	at net.runelite.client.util.RunnableExceptionLogger.run(RunnableExceptionLogger.java:41)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Unknown Source)
	at java.base/java.util.concurrent.FutureTask.run(Unknown Source)
	at java.base/java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
	at java.base/java.lang.Thread.run(Unknown Source)
Caused by: java.lang.RuntimeException: net.runelite.client.plugins.PluginInstantiationException: java.lang.IllegalStateException: must be called on client thread
	at net.runelite.client.externalplugins.ExternalPluginManager.lambda$refreshPlugins$3(ExternalPluginManager.java:344)
	at java.desktop/java.awt.event.InvocationEvent.dispatch(Unknown Source)
	at java.desktop/java.awt.EventQueue.dispatchEventImpl(Unknown Source)
	at java.desktop/java.awt.EventQueue$4.run(Unknown Source)
	at java.desktop/java.awt.EventQueue$4.run(Unknown Source)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(Unknown Source)
	at java.desktop/java.awt.EventQueue.dispatchEvent(Unknown Source)
	at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(Unknown Source)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(Unknown Source)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForHierarchy(Unknown Source)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(Unknown Source)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(Unknown Source)
	at java.desktop/java.awt.EventDispatchThread.run(Unknown Source)
Caused by: net.runelite.client.plugins.PluginInstantiationException: java.lang.IllegalStateException: must be called on client thread
	at net.runelite.client.plugins.PluginManager.startPlugin(PluginManager.java:460)
	at net.runelite.client.externalplugins.ExternalPluginManager.lambda$refreshPlugins$3(ExternalPluginManager.java:339)
	... 13 common frames omitted
Caused by: java.lang.IllegalStateException: must be called on client thread
	at dj.rt(dj.java:53119)
	at dj.getWorldLocation(dj.java:36644)
	at io.cbitler.stealingartefacts.StealingArtefactsPlugin.handleRefreshCurrentHouse(StealingArtefactsPlugin.java:284)
	at io.cbitler.stealingartefacts.StealingArtefactsPlugin.loadConfig(StealingArtefactsPlugin.java:301)
	at io.cbitler.stealingartefacts.StealingArtefactsPlugin.startUp(StealingArtefactsPlugin.java:88)
	at net.runelite.client.plugins.PluginManager.startPlugin(PluginManager.java:429)
	... 14 common frames omitted
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,5 +1,7 @@
 ## Unreleased
 
+- Fix: Issue with plugin not starting at startup. (#37)
+
 ## 1.5
 
 - Minor: Add an option for only showing unlured guards in the overlay. (#24)

--- a/src/main/java/io/cbitler/stealingartefacts/StealingArtefactsPlugin.java
+++ b/src/main/java/io/cbitler/stealingartefacts/StealingArtefactsPlugin.java
@@ -5,6 +5,7 @@ import lombok.extern.slf4j.Slf4j;
 import net.runelite.api.*;
 import net.runelite.api.coords.WorldPoint;
 import net.runelite.api.events.*;
+import net.runelite.client.callback.ClientThread;
 import net.runelite.client.config.ConfigManager;
 import net.runelite.client.eventbus.Subscribe;
 import net.runelite.client.events.ConfigChanged;
@@ -49,6 +50,9 @@ public class StealingArtefactsPlugin extends Plugin {
     private Client client;
 
     @Inject
+    private ClientThread clientThread;
+
+    @Inject
     private XpTrackerService xpTrackerService;
 
     public HashSet<GameObject> markedObjects = new HashSet<>();
@@ -85,7 +89,7 @@ public class StealingArtefactsPlugin extends Plugin {
 
         if (client.getGameState() == GameState.LOGGED_IN)
         {
-            loadConfig();
+            clientThread.invoke(this::loadConfig);
         }
     }
 
@@ -99,8 +103,12 @@ public class StealingArtefactsPlugin extends Plugin {
         this.overlayManager.remove(patrolOverlay);
         this.overlayManager.remove(khaledOverlay);
 
-        if (client.getLocalPlayer() != null && isInPisc(client.getLocalPlayer().getWorldLocation())) {
-            client.clearHintArrow();
+        if (client.getLocalPlayer() != null) {
+            clientThread.invoke(() -> {
+                if (isInPisc(client.getLocalPlayer().getWorldLocation())) {
+                    client.clearHintArrow();
+                }
+            });
         }
     }
 
@@ -131,7 +139,7 @@ public class StealingArtefactsPlugin extends Plugin {
         if (!c.getGroup().equalsIgnoreCase(StealingArtefactsConfig.GROUP_NAME)) {
             return;
         }
-        handleRefreshCurrentHouse();
+        clientThread.invoke(this::handleRefreshCurrentHouse);
     }
 
     /**

